src/span.js

```diff
@@ -180,6 +239,7 @@ export default class Span {
      *         implementation) will be used.
      */
     finish(finishTime) {
+        // Debug-only runtime checks on the arguments
```

Comment:
nit: other platforms also accept a collection of logs, but that can be a different PR.